### ISSUE TITLE
GUI of 7.6.8 version

### ISSUE THREAD

**Issue by thuzhf** [Hagb/docker-easyconnect#46]
I upgrade to the latest 7.6.8 version, and run the container, the following error exists:
```
...
Starting applications specified in flwm
Log file is /root/.vnc/3ee1fb5fdb28:1.log

Use xtigervncviewer -SecurityTypes VncAuth,TLSVnc -passwd /root/.vnc/passwd 3ee1fb5fdb28:1 to connect to the VNC server.

/usr/local/bin/start-sangfor.sh: line 15: /usr/share/sangfor/EasyConnect/EasyConnect: No such file or directory
/usr/local/bin/start-sangfor.sh: line 15: /usr/share/sangfor/EasyConnect/EasyConnect: No such file or directory
/usr/local/bin/start-sangfor.sh: line 15: /usr/share/sangfor/EasyConnect/EasyConnect: No such file or directory
...
```

**Comment by Hagb** [comment]
How did you build the container? https://github.com/shmilee/scripts/releases/download/v0.0.1/easyconn_7.6.8.2-ubuntu_amd64.deb is only for cli container.

**Comment by thuzhf** [comment]
Sorry, I thought it also worked for the graphical version. Then how can I use 7.6.8 in graphical version? because my organization uses the latest version for the server, and if the client version doesn't match the server, the client will report "version not match error" and won't work.

**Comment by Hagb** [comment]
Try to cover the config file (`/usr/share/sangfor/EasyConnect/resources/conf/`) and some binaries in `7.6.7` container by the `7.6.8` cli ones (I write a Dockerfile to do that):

``` Dockerfile
FROM hagb/docker-easyconnect:7.6.7

SHELL ["bash", "-c"]

RUN cd /tmp && \
    busybox wget https://hub.fastgit.org/shmilee/scripts/releases/download/v0.0.1/easyconn_7.6.8.2-ubuntu_amd64.deb -O easyconn.deb && \
    dpkg -x easyconn.deb easyconn && rm easyconn.deb && \
    cp easyconn/usr/share/sangfor/EasyConnect/resources/bin/{CSClient,svpnservice,ca.crt,cert.crt} \
               /usr/share/sangfor/EasyConnect/resources/bin/ && \
    chmod +sx /usr/share/sangfor/EasyConnect/resources/bin/{CSClient,svpnservice} && \
    rm -r /usr/share/sangfor/EasyConnect/resources/conf && \
    cp -r easyconn/usr/share/sangfor/EasyConnect/resources/conf /usr/share/sangfor/EasyConnect/resources/conf && \
    rm -r easyconn

CMD start.sh
```
Save it to `Dockerfile.7.6.8.cli`, and then run
``` shell
docker build -f Dockerfile.7.6.8.gui -t docker-easyconnect:7.6.8.gui .
```
to build.

Finally, run
```
docker run --device /dev/net/tun --cap-add NET_ADMIN -ti -e PASSWORD=xxxx -p 127.0.0.1:5901:5901 -p 127.0.0.1:1080:1080  docker-easyconnect:7.6.8.gui
```
to start the container, and use the GUI in VNC (`127.0.0.1:5901`).

I don't know whether it will work. I tried to use `7.6.3` config files, and logined successfully, but GUI didn't display the correct status.

**Comment by thuzhf** [comment]
I tried your updated dockerfile.7.6.8.gui just now, but it still shows the following error after I input the address and click the enter key:
![image](https://user-images.githubusercontent.com/8064242/111118703-1a75f180-85a4-11eb-9ea3-dfb487721454.png)

**Comment by Hagb** [comment]
> I don't know whether it will work. I tried to use `7.6.3` config files, and logined successfully (and VNC also worked), but GUI didn't display the correct status.

I made a mistake about versions of these images, and the above method doesn't work in fact.
It seems that combination of GUI and configure files of different version is unuseable.

**Comment by ruher** [comment]
> Sorry, I thought it also worked for the graphical version.
> 
> Then how can I use 7.6.8 in graphical version? because my organization uses the latest version for the server, and if the client version doesn't match the server, the client will report "The client version and server software version is not matching. Download Updates" and won't work, and when I click the "Download Updates" nothing happens.

not sure if it is what you want, I just found that one can change the version value in  /EasyConnect/rersources/conf/Version.xml,  then it passes the version check

**Comment by Hagb** [comment]
Now I think `7.6.7` client is compatible with `7.6.8` and the so-call official `7.6.8` client may even not exist, because even the official `7.6.8R2` server provides only `7.6.7` client.
Try to use the latest image in dockerhub or to build it by yourself.

**Comment by RobinWu218** [comment]
> > Sorry, I thought it also worked for the graphical version.
> > Then how can I use 7.6.8 in graphical version? because my organization uses the latest version for the server, and if the client version doesn't match the server, the client will report "The client version and server software version is not matching. Download Updates" and won't work, and when I click the "Download Updates" nothing happens.
> 
> not sure if it is what you want, I just found that one can change the version value in /EasyConnect/rersources/conf/Version.xml, then it passes the version check

@ruher  Would you please elaborate more on how you did this? I modified the Version.xml while running the container but it still asks for updating. 

Specifically, I modified the xml file to be
`<?xml version="1.0" encoding="utf-8"?>
<root>
<alias>linux_768</alias>
<version>7.6.8</version>
<pkg>deb</pkg>
<arch>x64</arch>
</root>`

Did you do something different? Thanks a lot.

**Comment by ruher** [comment]
Actually I am not using container, maybe I should not comment here… but my
solution does work so you can have a try. Install the .deb package, solve
some library dependencies as described by many other users, then for the
first time when you login, no problem at all, once you disconnect and
reconnect you get the version problem, at that time you can modify the
.xml, please check another .xml file in the same folder where the details
of version comparison is clearly described. That’s all I know for now. Good
luck, and f*ck this stupid company!

**Comment by Hagb** [comment]
> @ruher Would you please elaborate more on how you did this? I modified the Version.xml while running the container but it still asks for updating.
> 
> Specifically, I modified the xml file to be `<?xml version="1.0" encoding="utf-8"?> <root> <alias>linux_768</alias> <version>7.6.8</version> <pkg>deb</pkg> <arch>x64</arch> </root>`
> 
> Did you do something different? Thanks a lot.

I think it has been solved by f4ef12f90b7a076ab1b7743de9887569e5ae277b and/or 8a1dd3b5644716dd9f664b23fa1bab2e4fcb3555 automatically. Now using the latest EasyConnect should just be fine, or (if for some reasons you have to use older client) there is also a workaround `-e DISABLE_PKG_VERSION_XML=1` which disables the version check.



> Actually I am not using container, maybe I should not comment here… but my solution does work so you can have a try. Install the .deb package, solve some library dependencies as described by many other users, then for the first time when you login, no problem at all, once you disconnect and reconnect you get the version problem, at that time you can modify the .xml, please check another .xml file in the same folder where the details of version comparison is clearly described. That’s all I know for now. Good luck, and f*ck this stupid company!

`ln -fs /dev/null /usr/share/sangfor/EasyConnect/resources/conf/pkg_version.xml` can disables the version check. (8a1dd3b5644716dd9f664b23fa1bab2e4fcb3555)
